e2etests/test/module_identity_helper.js

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+var uuid = require('uuid');
+var debug = require('debug')('e2etests:module-identity-helper');
+var async = require('async');
+
+var Registry = require('azure-iothub').Registry;
+var ConnectionString = require('azure-iot-common').ConnectionString;
+var DeviceClient = require('azure-iot-device').Client;
+var ServiceClient = require('azure-iothub').Client;
+var DeviceIdentityHelper = require('./device_identity_helper.js');
+
+var hubConnectionString = process.env.IOTHUB_CONNECTION_STRING;
+var registry = Registry.fromConnectionString(hubConnectionString);
+var serviceClient = ServiceClient.fromConnectionString(hubConnectionString);
+
+module.exports.createModule = function(testModule, Transport, done)  {
+  async.series([
+    function createDeviceIfNecessary(done) {
+      if (testModule.testDevice) {
+        done();
+      } else {
+        testModule.registry = registry;
+        testModule.serviceClient = serviceClient;
+
+        debug('Creating SAS device to use with test');
+        DeviceIdentityHelper.createDeviceWithSas(function (err, testDevice) {
```

Comment:
>DeviceIdentityHelper [](start = 8, length = 20)

needs to be renamed to just RegistryHelper or RegistryIndentityHelper but Device seems wrong now.